src/btreemap/allocator.rs

```diff
@@ -401,6 +413,40 @@ mod test {
         assert_eq!(allocator.num_allocated_chunks, 0);
     }
 
+    #[test]
+    fn allocate_then_clear() {
+        let mem = make_memory();
+        let allocation_size = Bytes::from(16u64);
+        let allocator_addr = Address::from(0);
+
+        let mut allocator = Allocator::new(mem.clone(), allocator_addr, allocation_size);
+
+        allocator.allocate();
+        allocator.allocate();
+
+        assert_eq!(
+            allocator.free_list_head,
+            allocator_addr
+                + AllocatorHeader::size()
+                + allocator.chunk_size()
+                + allocator.chunk_size()
+        );
+        allocator.clear();
+        assert_eq!(
+            allocator.free_list_head,
+            allocator_addr + AllocatorHeader::size()
+        );
+        assert_eq!(allocator.num_allocated_chunks, 0);
```

Comment:
What if we add new fields to the `Allocator` that need to also be reset on `clear`? If we forget to reset these fields, this test will still pass.

A more robust assertion would be to:
1. Read the `AllocatorHeader`
2. Create a `new` allocator and read its `AllocatorHeader`
3. Verify that headers in steps 1 and 2 are identical.